pyfr/h5writer.py

```diff
@@ -10,46 +10,48 @@
 from pyfr.mpiutil import get_comm_rank_root
 
 
-class H5Writer(BaseIntegrator):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+class H5Writer(object):
+    def __init__(self, intg, cfgsect, prefix):
+        cfg = intg.cfg
+        self.prefix = prefix
 
         # Base output directory and file name
-        self._basedir = self.cfg.getpath('soln-output', 'basedir', '.')
-        self._basename = self.cfg.get('soln-output', 'basename', raw=True)
+        self._basedir = cfg.getpath(cfgsect, 'basedir', '.')
```

Comment:
Decouple from the specifics of our config file format.  These should be parameters.